### PR TITLE
[BUG] test_all_rule_queries_optimized does not run on rules

### PR DESCRIPTION
**Pull request by eric-forte-elastic**
## Summary

The language filter to test if rule queries are optimized was searching for `kql` instead of `kuery` in the language data. This causes the optimization check to be skipped as no rules have `kql` as their language. 

## Testing 

Run the `test_all_rule_queries_optimized` unit test and see if it passes. If not rule updates may be necessary to make queries more optimized.

### CODE REVIEW

**Comment by terrancedejesus** [comment]
👋🏽 - 219 rules have queries that are not optimized and would cause unit tests to fail. These rules have been automatically updated with the code below. 

```python
from detection_rules.rule_loader import RuleCollection
rules = RuleCollection.default()
import dataclasses
import time
import kql
from detection_rules.rule import (AnyRuleData, BaseRuleData, DeprecatedRule, QueryRuleData,
                   ThreatMapping, TOMLRule, TOMLRuleContents, RuleTransform)
from detection_rules.version_lock import load_versions
from semver import Version

today = time.strftime("%Y/%m/%d")
version_lock = load_versions()
for rule in rules:
    if rule.contents.data.get("language") == "kuery":
        source = rule.contents.data.query
        tree = kql.parse(source, optimize=False)
        optimized = tree.optimize(recursive=True)
        if source.strip() != str(optimized):
            locks = version_lock.get(rule.contents.data.rule_id, {})
            if locks:
                previous_locks = locks.get("previous", {})
                if previous_locks:
                    for v in previous_locks.keys():
                        if Version.parse(v, optional_minor_and_patch=True) < \
                            Version.parse(rule.contents.metadata.get("min_stack_version")) \
                                and Version.parse(v, optional_minor_and_patch=True) >= Version.parse('8.3.0'):
                            print(f"{rule.contents.data.rule_id} - min-stack: {rule.contents.metadata.get('min_stack_version')} - forked: {v}")
            new_meta = dataclasses.replace(rule.contents.metadata, updated_date=today)
            new_data = dataclasses.replace(rule.contents.data, query=optimized)
            new_contents = dataclasses.replace(rule.contents, data=new_data, metadata=new_meta)
            new_rule = TOMLRule(contents=new_contents, path=rule.path)
            new_rule.save_toml()
```

Additionally, these are the rules that will potentially cause the previous branches to be in a failing state from unit tests. These rules all have a min-stack version that is higher that `8.3.0` but were previously forked at `8.3.0` or higher but less than the min-stack. As we know, all rules backport to at least `8.3.0` and since we are making a code change, it will backport to `7.16`+ branches. The potential failing unit tests may extend beyond the `8.3` branch, but with the recent nod to limit backports to `8.3.0`, we may reduce those potential failures significantly.

```console
a00681e3-9ed6-447c-ab2c-be648821c622 - min-stack: 8.6.0 - forked: 8.3
07b5f85a-240f-11ed-b3d9-f661ea17fbce - min-stack: 8.4.0 - forked: 8.3
9510add4-3392-11ed-bd01-f661ea17fbce - min-stack: 8.4.0 - forked: 8.3
495e5f2e-2480-11ed-bea8-f661ea17fbce - min-stack: 8.4.0 - forked: 8.3
7caa8e60-2df0-11ed-b814-f661ea17fbce - min-stack: 8.4.0 - forked: 8.3
a2795334-2499-11ed-9e1a-f661ea17fbce - min-stack: 8.4.0 - forked: 8.3
cad4500a-abd7-4ef3-b5d3-95524de7cfe1 - min-stack: 8.4.0 - forked: 8.3
5e161522-2545-11ed-ac47-f661ea17fbce - min-stack: 8.4.0 - forked: 8.3
68994a6c-c7ba-4e82-b476-26a26877adf6 - min-stack: 8.4.0 - forked: 8.3
a99f82f5-8e77-4f8b-b3ce-10c0f6afbc73 - min-stack: 8.4.0 - forked: 8.3
cc6a8a20-2df2-11ed-8378-f661ea17fbce - min-stack: 8.4.0 - forked: 8.3
e555105c-ba6d-481f-82bb-9b633e7b4827 - min-stack: 8.4.0 - forked: 8.3
12a2f15d-597e-4334-88ff-38a02cb1330b - min-stack: 8.4.0 - forked: 8.3
14de811c-d60f-11ec-9fd7-f661ea17fbce - min-stack: 8.4.0 - forked: 8.3
65f9bccd-510b-40df-8263-334f03174fed - min-stack: 8.4.0 - forked: 8.3
764c8437-a581-4537-8060-1fdb0e92c92d - min-stack: 8.4.0 - forked: 8.3
12cbf709-69e8-4055-94f9-24314385c27e - min-stack: 8.4.0 - forked: 8.3
df7fda76-c92b-4943-bc68-04460a5ea5ba - min-stack: 8.4.0 - forked: 8.3
2abda169-416b-4bb3-9a6b-f8d239fd78ba - min-stack: 8.4.0 - forked: 8.3
c7908cac-337a-4f38-b50d-5eeb78bdb531 - min-stack: 8.4.0 - forked: 8.3
```

File: tests/test_all_rules.py [hunk 1]
```diff
@@ -58,7 +58,7 @@ def test_file_names(self):
     def test_all_rule_queries_optimized(self):
         """Ensure that every rule query is in optimized form."""
         for rule in self.production_rules:
-            if rule.contents.data.get("language") == "kql":
+            if rule.contents.data.get("language") == "kuery":
```
Comment (brokensound77) [inline tests/test_all_rules.py:61]: nice find - That has been there a while

**Review by brokensound77 — CHANGES_REQUESTED** [review]
Nice find and fix. Is the whitespace flattening forced by kql optimize, or was that from automation? I would say it only makes sense with whitespace insensitivity, because it takes more away from readability than it adds.

So, if it was forced by optimize, we should see about removing the whitespace enforcement or else remove this as a unit test

**Comment by eric-forte-elastic** [comment]
@brokensound77 would like @terrancedejesus to also confirm, but yes the whitespace enforcement is part of optimize. 

For example the diff for `rules/cross-platform/initial_access_zoom_meeting_with_no_passcode.toml` is just a whitespace flattening and is the raw result from optimize. 

image

Confirmed through the edit of the `test_all_rule_queries_optimized`:

```python
    def test_all_rule_queries_optimized(self):
        """Ensure that every rule query is in optimized form."""
        for rule in self.production_rules:
            if "initial_access_zoom_meeting_with_no_passcode" in str(rule.path):    
                if rule.contents.data.get("language") == "kuery":
                    source = rule.contents.data.query
                    tree = kql.parse(source, optimize=False)
                    optimized = tree.optimize(recursive=True)
                    err_message = f'\n{self.rule_str(rule)} Query not optimized for rule\n' \
                                f'Expected: {optimized}\nActual: {source}'
                    self.assertEqual(tree, optimized, err_message)
```
Source value: 
`'event.type:creation and event.module:zoom and event.dataset:zoom.webhook and\n  event.action:meeting.created and not zoom.meeting.password:*\n'`

Optimized value:
`'event.type:creation and event.module:zoom and event.dataset:zoom.webhook and event.action:meeting.created and not zoom.meeting.password:*'` 

<details><summary>Debugging Screenshot</summary>
<p>

image


</p>
</details>

**Comment by brokensound77** [comment]
>@brokensound77 would like @terrancedejesus to also confirm, but yes the whitespace is part of optimize, but not enforced in the unit test.

Thanks @eric-forte-elastic 

I think that means that we _can_ leave that unit test in with the fix, but would need to make the rule changes by hand to preserve existing whitespace. Since it has not been working for some time, we could also just choose to comment it out until some time in the future

**Review by Mikaayenson — APPROVED** [review]
🔥 great work!

**Review by terrancedejesus — APPROVED** [review]
Changes look good. Since we limited backports, most of the rule versions are no longer forked so we shouldn't run into any issues with backporting.

**Pull request merged**